include/fem-dem/ib_particles_dem.h

```diff
@@ -145,6 +160,40 @@ class IBParticlesDEM
                              std::vector<Tensor<1, 3>> &contact_torque);
 
 
+  /**
+   * @brief Calculates non-linear (Hertzian) force between two objects. Take into input the physical properties and contact state of the two objects.
+   */
+  void
+  calculate_force_model(const double  normal_overlap,
+                        ContactInfo  &contact_info,
+                        Point<3>     &contact_point,
+                        Tensor<1, 3> &contact_normal,
+                        Tensor<1, 3> &normal_force,
+                        Tensor<1, 3> &tangential_force,
+                        Tensor<1, 3> &rolling_resistance_torque,
+                        Point<dim>   &particle_one_position,
+                        Tensor<1, 3> &particle_one_velocity,
+                        Tensor<1, 3> &particle_one_omega,
+                        const double  particle_one_mass,
+                        const double  particle_one_radius,
+                        const double  particle_one_youngs_modulus,
+                        const double  particle_one_poisson_ratio,
+                        const double  particle_one_restitution_coefficient,
+                        const double  particle_one_friction_coefficient,
+                        const double  particle_one_rolling_friction_coefficient,
+                        Point<dim>   &particle_two_position,
+                        Tensor<1, 3> &particle_two_velocity,
+                        Tensor<1, 3> &particle_two_omega,
+                        const double  particle_two_mass,
+                        const double  particle_two_radius,
+                        const double  particle_two_youngs_modulus,
+                        const double  particle_two_poisson_ratio,
+                        const double  particle_two_restitution_coefficient,
+                        const double  particle_two_friction_coefficient,
+                        const double  particle_two_rolling_friction_coefficient,
+                        const double  dt);
```

Comment:
Ok this is a mess to have a function with so many argument
Why not make a struct that contains all this information and just pass the struct? Otherwise it becomes a bit unreadable to know what is what. I would have maybe a struct per particle one. Like struct contact_info and have one for particle 1 and one for particle 2.